gammapy/modeling/models/spatial.py

```diff
@@ -77,7 +77,7 @@ def position_error(self):
                 angle=np.nan * u.deg,
             )
         pars = self.parameters
-        idx = [pars._get_idx("lon_0"), pars._get_idx("lat_0")]
+        idx = [pars._get_idx(name) for name in ["lon_0", "lat_0"]]
```

Comment:
Shouldn't the idea here be that you don't need to work with `idx` externally at all?
It should be possible to call a method to get the sub-covar directly?
Doesn't this work already or should work?
```
sub_covar = pars.get_covariance(["lon_0", "lat_0"])
```

Reply:
If you don't want pars._get_idx("lon_0") here, we need a method to get sub-covariance by nam. Axel said it's not safe because of parameters duplication so I removed it.

Reply:
Maybe that was a misunderstanding, my [comment](https://github.com/gammapy/gammapy/pull/2522#discussion_r343736895) was that `parameters.get_subcovariance()` should not document the possibility to call it with names, as it is un-safe. Internally (as you know, that there are no parameter duplications) it still work and you can use it...